src/npc.h

```diff
@@ -943,4 +943,7 @@ struct epilogue {
     epilogue* find_epilogue(std::string ident);
     void random_by_group(std::string group, std::string name);
 };
+
+std::ostream& operator<< (std::ostream & os, npc_need need);
```

Comment:
With smaller values, e.g. ints, pass by value tends to be faster, so I do that by default. ..not that it would makes much difference here. I figure it'd be optimized to a value sitting in a register, whereas pass-by-reference involves dereferencing. Even for larger type, pass-by-value can be faster in some cases e.g. with RVO optimization. There is an old article, "Want speed? Pass by value" that goes in to some of this, fyi.

Let me know if you strongly feel it should be pass-by-ref-to-const, and I'll change it.